### ISSUE TITLE
Add SARIF support

### ISSUE THREAD

**Issue by mathroule** [adrienverge/yamllint#578]
It would be nice to be able to format output as [SARIF](https://docs.oasis-open.org/sarif/sarif/v2.1.0/sarif-v2.1.0.html).

**Comment by cheesymoon** [comment]
No native SARIF yet, but until (if) it lands: https://github.com/sarif-kit/sarif-kit converts `yamllint -f parsable` output to SARIF 2.1.0. CLI on PyPI, or container action:

```yaml
- run: pipx run yamllint -f parsable . > yamllint.txt || [ $? -eq 1 ]
- uses: sarif-kit/sarif-kit@v0.2.0
  with:
    tool: yamllint
    input: yamllint.txt
    output: yamllint.sarif
- uses: github/codeql-action/upload-sarif@v4
  with:
    sarif_file: yamllint.sarif
    category: yamllint
```

Parsable is the only format it takes, being the one that keeps line, column, level and rule id on every line. Those all carry over. Rule id becomes the SARIF rule and links to its page in the yamllint docs, which is what GitHub builds the alert title from.

yamllint exits 1 on lint errors. Guard swallows that and nothing else, so bad flag or missing binary still fails the job.

Working example: [sarif-kit/demo](https://github.com/sarif-kit/demo), broken on purpose, its [yamllint workflow](https://github.com/sarif-kit/demo/blob/master/.github/workflows/yamllint.yml) runs the snippet above. Alerts look like this: https://sarif-kit.github.io/sarif-kit/yamllint/
